### PR TITLE
tso: refine the allocator setup method and tests

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

* Refine the allocator setup methods.
* Refine and speed up the tests.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

**Review by Yisaer — APPROVED** [review]
LGTM

File: tests/server/tso/manager_test.go [hunk 1]
```diff
@@ -237,5 +212,5 @@ func waitAllocatorPriorityCheck(cluster *tests.TestCluster) {
 	wg.Wait()
 	// Because the leader changing may take quite a long period,
 	// so we sleep longer here to wait.
-	time.Sleep(waitAllocatorPriorityCheckInterval)
+	time.Sleep(time.Minute)
```
Comment (rleungx) [inline tests/server/tso/manager_test.go:215]: can we make it shorter?

Comment (JmPotato) [reply tests/server/tso/manager_test.go:215]: I think it could be shorter after #3387 merges.

Comment (JmPotato) [reply tests/server/tso/manager_test.go:243]: After some testing, it turns out we can't make this become shorter...

File: pkg/testutil/testutil.go [hunk 2]
```diff
@@ -34,30 +34,38 @@ type CheckFunc func(c *check.C) bool
...
-	options.SleepInterval = waitRetrySleep
+	option := &WaitOp{
+		retryTimes:    waitMaxRetry,
+		sleepInterval: waitRetrySleep,
+	}
 	for _, opt := range opts {
-		opt(options)
+		opt(option)
 	}
 	for i := 0; i < waitMaxRetry; i++ {
```
Comment (HunDunDM) [inline pkg/testutil/testutil.go:64]: Need to use `option.retryTimes` here?

**Review by HunDunDM — COMMENTED** [review]
rest LGTM

File: pkg/testutil/testutil.go (hunk 2 above)
Comment (JmPotato) [reply pkg/testutil/testutil.go]: Fixed. PTAL again.

**Review by rleungx — APPROVED** [review]

**Pull request merged**